platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencySubstitutionRulesIntegrationTest.groovy

```diff
@@ -1612,4 +1658,64 @@ Required by:
             }
         }
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/7594")
+    def "can substitute module with project and use withoutArtifactSelectors"() {
```

Comment:
Can this be made to use the resolve fixture to verify the result too?

Reply:
Done